### PR TITLE
Migrate one test from compiler to regular code

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: poko-tests/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  tvosArm64()
+  tvosX64()
+  tvosSimulatorArm64()
+
+  watchosArm32()
+  watchosArm64()
+  watchosDeviceArm64()
+  watchosSimulatorArm64()
+  watchosX64()
```
Comment (JakeWharton) [inline poko-tests/build.gradle.kts:32]: To be de-duplicated later. Alternatively, the test subjects and the tests could live entirely within the annotation module's `commonTest/` folder.

**Comment by JakeWharton** [comment]
The reason this works on all targets is that the test still passes even when Poko compiler plugin does not run. The next test, once the structure is approved, will require the compiler plugin and will contain the updates to ensure it runs on all targets.

File: poko-tests/src/commonTest/kotlin/ExplicitDeclarationsTest.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotEqualTo
+import kotlin.test.Test
+
+class ExplicitDeclarationsTest {
+    @Test fun twoEquivalentCompiledExplicitDeclarationsInstancesAreEquals() {
```
Comment (JakeWharton) [inline poko-tests/src/commonTest/kotlin/ExplicitDeclarationsTest.kt:7]: JS can't do spaces in function names

File: poko-tests/src/commonMain/kotlin/ExplicitDeclarations.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    override fun hashCode() = string.length
+}
+
+data class ExplicitDeclarationsData(
+    private val string: String
+) {
+    override fun toString() = string
+    override fun equals(other: Any?) = other is ExplicitDeclarationsData && other.string.length == string.length
+    override fun hashCode() = string.length
+}
```
[the remark this answers is not in the document]
Comment (drewhamilton) [reply poko-tests/src/commonMain/kotlin/ExplicitDeclarations.kt:17]: Yeah, I prefer the clarity of writing them separately.

File: poko-tests/src/commonMain/kotlin/ExplicitDeclarations.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+@Poko class ExplicitDeclarationsPoko(
+    private val string: String
+) {
+    override fun toString() = string
+    override fun equals(other: Any?) = other is ExplicitDeclarationsPoko && other.string.length == string.length
+    override fun hashCode() = string.length
+}
+
+data class ExplicitDeclarationsData(
```
Comment (drewhamilton) [inline poko-tests/src/commonMain/kotlin/ExplicitDeclarations.kt:11]: The different names work fine for this test, but for the typical-case `toString` tests the class names can't match unless they're in different packages.

**Review by drewhamilton — APPROVED** [review]

File: poko-tests/src/commonMain/kotlin/ExplicitDeclarations.kt (hunk 4 above)
Comment (JakeWharton) [reply poko-tests/src/commonMain/kotlin/ExplicitDeclarations.kt:11]: I'll switch to import aliases

**Pull request merged**